examples/autoignition-mpi.py

```diff
@@ -52,73 +55,103 @@
 from mirgecom.boundary import AdiabaticSlipBoundary
 from mirgecom.initializers import MixtureInitializer
 from mirgecom.eos import PyrometheusMixture
+
+from mirgecom.logging_quantities import (
+    initialize_logmgr,
+    logmgr_add_many_discretization_quantities,
+    logmgr_add_device_name,
+    logmgr_add_device_memory_usage,
+    set_sim_state
+)
+
 import cantera
 import pyrometheus as pyro
 
 logger = logging.getLogger(__name__)
 
 
 @mpi_entry_point
-def main(ctx_factory=cl.create_some_context, casename="autoignition", use_leap=False,
-         restart_step=None, restart_name=None):
+def main(ctx_factory=cl.create_some_context, use_logmgr=False,
+         use_leap=False, use_profiling=False, casename="autoignition",
+         rst_step=None, rst_name=None):
```

Comment:
What's the difference between `casename` and `rst_name`?

Reply:
Restart name is user-specified restart file root, and casename is the name that the simulation will use to write the restart files.  They can be different.

Reply:
Hmm, still not sure I'm getting it (maybe because it's not currently being used in `autoignition-mpi`). Is it the same as `rst_path` then, or something different?

Reply:
Yeah, I may have flubbed up the translation of it to `autoignition` - I do realize that is sort of a hodge-podge of stuff for restart because I was trying to help roll some support for the stuff being done in production into the examples.

So... sometimes the restart file that you specify as the file to currently restart from is not named the same as what your simulation is currently writing as restarts.  You will want to be able to specify *any* filename to restart from.  This is user-defined.

On the other hand, the simulation will always write the restart data at restart_path/casename-<stamp>.rst.   But it is not required that the file specified by the user at restart time (i.e. the one used for the actual restart) be named exactly like that, they can name it whatever they want.